src/odemis/util/img.py

```diff
@@ -811,3 +813,105 @@ def getBoundingBox(content):
+            data = numpy.add(img[0], img[1], dtype=self._best_dtype)
+            # update the metadata of the integrated image in every integration step
+            md = self.update_integration_metadada(md)
+            self._img = [model.DataArray(data, md)]
+
+            # At the end of the acquisition, check if the detector type is DT_NORMAL and then take the average by
+            # dividing with the number of acquired images (integration count) for every pixel position and restoring
+            # the original dtype.
+            if self._step == self.steps:
+                det_type = md.get(model.MD_DET_TYPE, model.MD_DT_INTEGRATING)
+                if det_type == model.MD_DT_NORMAL:  # SEM
+                    orig_dtype = img[-1].dtype
+                    data = numpy.divide(data, self.steps, dtype=orig_dtype)
+                if not det_type == model.MD_DT_INTEGRATING:  # optical
+                    logging.warning("Unknown detector type %s for image integration. "
+                                    "Image integration was performed anyways.", det_type)
+                # The baseline, if exists, should also be subtracted from the integrated image.
+                if model.MD_BASELINE in md:
+                    data, md = self.subtract_baseline(data, md)
+                # reset the ._img once you reach the integration count
+                self._img = None
+                self._step = 0
+
+            return [model.DataArray(data, md)]
+
+    def update_integration_metadada(self, md):
+        """
+        Update the metadata of each integrated image.
+        """
+        if MD_DWELL_TIME in md:
+            md[model.MD_DWELL_TIME] *= self._step  # total time ebeam stayed on same pixel/position
+        if MD_EXP_TIME in md:
+            md[model.MD_EXP_TIME] *= self._step
+        md[model.MD_INTEGRATION_COUNT] = self._step
+
+        return md
+
+    def subtract_baseline(self, data, md):
+        """
+        Subtract accumulated baselines from the data so that the data only has one.
+        Args:
+            data: the final data after integrating the acquired images
+            md: metadata of the integrated image
+        Returns:
+            data, md: the updated data and metadata after the subtraction of the baseline
+        """
+        baseline = model.MD_BASELINE
+        # Subtract the baseline (aka black level) from the final integrated image.
+        # Remove the baseline from n-1 images, keep one baseline as bg level.
+        minv = float(data.min())
+
+        # If the baseline is too high compared to the actual black, we could end up subtracting too much,
+        # and values would underflow => be extra careful and never subtract more than the min value.
+        baseline_sum = self.steps * baseline  # sum of baselines for n images.
+        extra_bl = (self.steps - 1) * baseline  # sum of baselines for n-1 images, keep one baseline as bg level.
```

Comment:
Use self._step, which is the actual number of integrations. In general it should be the same value, but if .steps is suddenly decreased, ._steps will still contain the right value.